webapp/src/pages/EditChatroomMemberPage/EditChatroomMemberPage.tsx

```diff
@@ -1,86 +1,207 @@
+  const { navigate } = useNavigation();
+  const getChatroom = useCallback(
+    () => chatApi.chatControllerGetChatroomById({ id: chatroomId! }),
+    [chatroomId],
+  );
+  const { data: chatroom, isLoading: isChatroomLoading } =
+    useData<Chatroom>(getChatroom);
+  const getUserByUserName = useCallback(
+    () => usersApi.userControllerGetUserByUserName({ userName: username! }),
+    [username],
+  );
+  const { data: user, isLoading: isUserLoading } = useData(getUserByUserName);
+  const useGetChatroomMember = (id?: string) =>
+    useCallback(
+      () =>
+        chatApi.chatControllerGetChatroomMember({
+          chatroomId: chatroomId!,
+          userId: id ?? '',
+        }),
+      [id],
+    );
+  const { data: crMember, isLoading: isCrMemberUserLoading } = useData(
+    useGetChatroomMember(user?.id),
+  );
+
+  const { authUser, isLoading: isAuthUserLoading } = useAuth();
+  const { data: authCrMember, isLoading: isAuthCrMemberUserLoading } = useData(
+    useGetChatroomMember(authUser?.id),
+  );
+
+  const isAuthOwner = authUser?.id === chatroom?.ownerId ?? false;
+  const isAuthAdmin = authCrMember?.admin ?? false;
+  const isAuthBanned = authCrMember?.banned ?? true;
+  const isDestOwner = chatroom?.ownerId === user?.id ?? true;
+  const isDestAdmin = crMember?.admin ?? true;
+
+  const canEdit =
+    (isAuthOwner || (isAuthAdmin && !isAuthBanned && !isDestAdmin)) &&
+    !isDestOwner;
+
+  const removeChatMember = useCallback(async () => {
+    try {
+      await chatApi.chatControllerRemoveChatroomMember({
+        chatroomId: chatroomId!,
+        userId: user?.id ?? '',
+      });
+      navigate(`${CHATROOM_URL}/${chatroomId}/details`);
+    } catch (error: unknown) {
+      if (error instanceof ResponseError) {
+        const responseBody = await error.response.json();
+        if (responseBody.message) {
+          warn(responseBody.message);
+        } else {
+          warn(error.response.statusText);
+        }
+      } else if (error instanceof Error) {
+        warn(error.message);
+      } else {
+        warn('Could not remove the chat member');
+      }
+    }
+  }, [chatroomId, user, navigate, warn]);
+  const [updateChatroomMember, setUpdateChatroomMember] =
+    useState<UpdateChatroomMemberDto | null>(null);
+  useEffect(() => {
+    if (!crMember) return;
+    setUpdateChatroomMember({
+      admin: crMember.admin,
+      muted: crMember.muted,
+      banned: crMember.banned,
+    });
+  }, [crMember]);
+  const genericOnToggle = (dto: UpdateChatroomMemberDto): (() => void) => {
+    return async () => {
+      if (!user || !chatroomId || !canEdit) {
+        if (!canEdit) warn("You can't modify this setting");
+        return;
+      }
+      try {
+        if (!isAuthOwner && dto.admin !== undefined) {
+          warn('You cannot make new admins');
+          return;
+        }
+        const oldUpdateChatroomMember = updateChatroomMember;
+        await chatApi.chatControllerUpdateChatroomMember({
```

Comment:
I think we can afford to make one api call per setting, instead of setting it locally and submitting all at once. Like that we don't need additional buttons, and we can validate more easily on the fly.